graylog2-web-interface/src/components/common/URLWhitelistFormModal.jsx

```diff
@@ -0,0 +1,158 @@
+// @flow strict
+import React from 'react';
+import PropTypes from 'prop-types';
+import uuid from 'uuid/v4';
+import connect from 'stores/connect';
+import { IfPermitted } from 'components/common';
+import { Button } from 'components/graylog';
+import BootstrapModalForm from 'components/bootstrap/BootstrapModalForm';
+import UrlWhiteListForm from 'components/configurations/UrlWhiteListForm';
+import CombinedProvider from 'injection/CombinedProvider';
+import PermissionsMixin from 'util/PermissionsMixin';
+import type { WhiteListConfig } from 'stores/configurations/ConfigurationsStore';
+
+const { CurrentUserStore } = CombinedProvider.get('CurrentUser');
+const { ConfigurationsActions, ConfigurationsStore } = CombinedProvider.get('Configurations');
+
+const URL_WHITELIST_CONFIG = 'org.graylog2.system.urlwhitelist.UrlWhitelist';
+
+type State = {
+  config: WhiteListConfig,
+  isValid: boolean
+};
+
+type Props = {
+  newUrlEntry: string,
+  onUpdate: () => void,
+  configuration: {},
+  currentUser: {permissions: Array<string>},
+  urlType: string,
+};
+
+
+class URLWhitelistFormModal extends React.Component<Props, State> {
+    configModal: ?BootstrapModalForm = React.createRef();
+
+    inputs = {};
+
+    static defaultProps = {
+      newUrlEntry: '',
+      onUpdate: () => {},
+      configuration: {},
+      urlType: '',
+    }
+
+    constructor(props) {
+      super(props);
+      this.state = {
+        config: { entries: [], disabled: false },
```

Comment:
For what I could understand about this component, `URLWhitelistFormModal` is holding the current state of the whitelist entries, but those are generated from props on render time and they only come into the state when `URLWhitelistForm`'s `onUpdate` is called.

That works, but I think it goes against _the React way_ of using a one-way data flow in your component hierarchy. I think initializing and keeping an up-to-date `config` state here would help us not relying on a child component updating to get the right state in this component.